packages/input/src/Input.js

```diff
@@ -21,12 +21,21 @@ Wrapper.propTypes = {
 };
 
 function Input(props) {
+  const {
+    onFocus: onFocusProp,
+    onBlur: onBlurProp,
+    onMouseEnter: onMouseEnterProp,
+    onMouseLeave: onMouseLeaveProp,
+    disabled: disabledProp,
```

Comment:
The tedious part of this approach is finding names for things our behavior components also use and provide. The `onFocus` prop a developer passes in is passed to the `ControlBehavior` component. The `ControlBehavior` component provides a different function also called `onFocus`. We pass that one through to the presenter. 

Potential for confusion or subtle bugs here I'm afraid.